File: src/main/java/so/aporia/utils/events/impl/RenderHudEvent.java
package so.aporia.utils.events.impl;

import net.minecraft.client.gui.GuiGraphics;

/** Fired each frame when the HUD is being rendered. */
public final class RenderHudEvent {

    private final GuiGraphics graphics;
    private final float       partialTick;

    public RenderHudEvent(GuiGraphics graphics, float partialTick) {
        this.graphics    = graphics;
        this.partialTick = partialTick;
    }

    public GuiGraphics graphics()    { return graphics; }
    public float       partialTick() { return partialTick; }
}

File: src/so/aporia/utils/events/impl/RenderHudEvent.kt
package so.aporia.utils.events.impl
import net.minecraft.client.gui.GuiGraphicsExtractor
import com.chaos.annotation.ChaosNative
@ChaosNative
class RenderHudEvent(val graphics: GuiGraphicsExtractor, val partialTick: Float) {
    fun graphics(): GuiGraphicsExtractor = graphics
    fun partialTick(): Float = partialTick
}